src/pathways/escher.service.ts

```diff
@@ -1,100 +1,135 @@
 import * as escher from 'escher-vis'
 import * as d3 from 'd3';
+import * as _ from 'lodash';
+import * as fp from 'lodash/fp';
 import 'jquery';
 
+/**
+ * Plan: the current structure is the following:
+ * pathway: {
+ *   reactions <= sorted reactions
+ *   primary_nodes <= the primary nodes belonging to each reaction
+ *   model <= model that ALSO contains reactions, but in the different order
+ * }
+ * I think reactions and primary_nodes should be merged together
+*/
+
 export class EscherService {
-	direction: number;
-	startCoordinates: any;
-	options: any;
-	metabolitesNames: any[];
-	reactions: any[];
-	self: EscherService;
+    direction: number;
+    options: any;
+    metabolitesNames: any[];
+    reactions: any[];
+
+    constructor() {
+        // TODO make it const
+        this.direction = -90;
+        this.metabolitesNames = [];
+        this.reactions = [];
+        this.options = {
+            // just show the zoom buttons
+            menu: 'zoom',
+            // do not use the smooth pan and zoom option
+            use_3d_transform: false,
+            // no editing in this map
+            enable_editing: true,
+            // show the descriptive names
+            identifiers_on_map: 'name',
+            // hide secondary metabolites
+            hide_secondary_metabolites: true,
+            // don't ask before quiting
+            never_ask_before_quit: true,
+            // disable keyboard shortcuts
+            enable_keys: false,
+            first_load_callback: this._firstLoadCallbackClosure(),
+        }
+    };
+
+    _firstLoadCallbackClosure() {
+        let self = this;
+        return function () {
+            // Using function expression because it needs to access the
+            // Builder's context as this
+            const startCoordinates = {x: 0, y: (self.reactions.length/2)*400 };
+
+            // So this is the second place where I'm zipping these together.
+            // Let's change the datastructure.
```

Comment:
Before changing things, ensure that you have tests covering the previous implementation. Only then do the changes.